### PR TITLE
implement `--agent` for `net peers`

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
## Summary of changes

Changes introduced in this pull request:

- added support for `Filecoin.NetAgentVersion` RPC command,
- added `--agent/-a` flag to `forest-cli net peers` subcommand.

Sample output:

```
❯ forest-cli net peers --agent
12D3KooWQPYouEAsUQKzvFUA9sQ8tz4rfpqtTzh2eL6USd9bwg7x, [/dns4/calibration.node.glif.io/tcp/1237/p2p/12D3KooWQPYouEAsUQKzvFUA9sQ8tz4rfpqtTzh2eL6USd9bwg7x], lotus-1.25.1+calibnet+git.1e0c8bb
12D3KooWJzQZHgMkMWipdUGKfBrnBbvi7UiC3pp8TQAWGrnNMNvF, [/ip4/52.14.121.29/tcp/34000/p2p/12D3KooWJzQZHgMkMWipdUGKfBrnBbvi7UiC3pp8TQAWGrnNMNvF], forest-0.16.7+git.1f8c947
12D3KooWS3ZRhMYL67b4bD5XQ6fcpTyVQXnDe8H89LvwrDqaSbiT, [/dns4/bootstrap-calibnet-1.chainsafe-fil.io/tcp/34000/p2p/12D3KooWS3ZRhMYL67b4bD5XQ6fcpTyVQXnDe8H89LvwrDqaSbiT], forest-0.16.6+git.fbd880d
12D3KooWEiBN8jBX8EBoM3M47pVRLRWV812gDRUJhMxgyVkUoR48, [/dns4/bootstrap-calibnet-2.chainsafe-fil.io/tcp/34000/p2p/12D3KooWEiBN8jBX8EBoM3M47pVRLRWV812gDRUJhMxgyVkUoR48], forest-0.16.6+git.fbd880d
12D3KooWLzM72EJeuXtfrpRnthkrGiGGDzGPF7C2ywbQmsqQzwmW, [/ip4/14.198.182.246/tcp/35167/p2p/12D3KooWLzM72EJeuXtfrpRnthkrGiGGDzGPF7C2ywbQmsqQzwmW], lotus-1.25.1+calibnet+git.eeccfcf0c.dirty
12D3KooWABQ5gTDHPWyvhJM7jPhtNwNJruzTEo32Lo4gcS5ABAMm, [/dns4/bootstrap-calibnet-0.chainsafe-fil.io/tcp/34000/p2p/12D3KooWABQ5gTDHPWyvhJM7jPhtNwNJruzTEo32Lo4gcS5ABAMm], forest-0.16.6+git.fbd880d
```

Then, we can get some basic and depressing stats for a single mainnet node:
```
❯ forest-cli net peers --agent | grep -i venus | wc -l
2
❯ forest-cli net peers --agent | grep -i lotus | wc -l
183
❯ forest-cli net peers --agent | grep -i forest | wc -l
1
```

```
     Running `target/debug/forest-tool api compare --filter Net`
| RPC Method               | Forest | Lotus |
|--------------------------|--------|-------|
| Filecoin.NetAddrsListen  | Valid  | Valid |
| Filecoin.NetAgentVersion | Valid  | Valid |
| Filecoin.NetPeers        | Valid  | Valid |
```

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by lemmih — COMMENTED** [review]
Looks good. Let's add a test in `src/tool/subcommands/api_cmd.rs`.

File: src/libp2p/service.rs [hunk 1]
```diff
@@ -548,6 +549,21 @@ async fn handle_network_message(
...
                     }
                 }
+                NetRPCMethods::AgentVersion(response_channel, peer_id) => {
+                    let agent_version = swarm
+                        .behaviour()
+                        .peer_info(&peer_id)
+                        .map(|info| {
+                            info.agent_version
+                                .clone()
+                                .unwrap_or_else(|| "item not found".to_string())
```
Comment (hanabi1224) [inline src/libp2p/service.rs:559]: I guess we can omit `unwrap_or_else` here if the channel takes `Option<String>`, and let the caller (which is more on UX side) to decide the fallback string

Comment (LesnyRumcajs) [reply src/libp2p/service.rs:559]: Let's keep the fallback the same as in Lotus.

**Comment by LesnyRumcajs** [comment]
> Looks good. Let's add a test in `src/tool/subcommands/api_cmd.rs`.

Done.

**Review by lemmih — APPROVED** [review]

**Pull request merged**